### DISCUSSION TITLE
Color pixel

### DISCUSSION THREAD

**Discussion by Bkeinn** [Textualize/rich#2087] [Q&A] [answered]
Hi, how do I add these colored pixels/blocks like in the preview to my code.
:smile:

**Comment by USLTD** [comment, marked as answer]
What do you mean **colored pixel/blocks**? Do you mean those in color palette?
![IMG_20220320_185444](https://user-images.githubusercontent.com/57063715/159168454-7d74bb10-424d-40ec-8c81-e5dfe9f18d0e.png)

If yes then you can use **RichRenderable** object for that https://github.com/Textualize/rich/blob/master/rich/__main__.py#L18-L36
It is constructed by using <kbd>▄</kbd> (*half-height pixel/cell*) character and carefully setting it's foreground and background which creates 2 color in the same terminal cell

**Comment by FedeClaudi** [comment]
Hey @UltraStudioLTD  thank's for posting that reply. Using the half height pixel with a background is such a clever way to create a beautiful, compact, colors display!

I was able to replicated in  [Term](https://github.com/FedeClaudi/Term.jl)

<img width="593" alt="image" src="https://user-images.githubusercontent.com/17436313/161379391-520a42bb-b7ec-4cbe-8c62-c68ffc7a361b.png">

**Reply by USLTD** [reply]
👍
